Tests/test_CodonUsage.py

```diff
@@ -28,3 +28,45 @@
 
 print("-" * 60)
 print("codon adaptation index for test gene: %.2f" % X.cai_for_gene("ATGAAACGCATTAGCACCACCATTACCACCACCATCACCATTACCACAGGTAACGGTGCGGGCTGA"))
+
+
+
+""" nRCA test
+    Creates a normRelativeCodonAdaptationIndex object, generates an index from
+    a reference set of genes and scores a set of genes (inputfile)
+    provided all in FASTA format
+"""
+# Copyright Ivan Erill 2018 (erill@umbc.edu)
+# -*- coding: utf-8 -*-
+from Bio import SeqIO  # To parse a FASTA file 
+
+# first make a nRCA object
+nrca_index = CodonUsage.normRelativeCodonAdaptationIndex()
+# now generate an index from a file
+if os.path.exists("./CodonUsage/refset_Sharp.fas"):
+    nrca_index.generate_index("./CodonUsage/refset_Sharp.fas")
+elif os.path.exists("./Tests/CodonUsage/refset_Sharp.fas"):
+    nrca_index.generate_index("./Tests/CodonUsage/refset_Sharp.fas")
+else:
+    print("Cannot find the file refset_Sharp.fas\n \
+          Make sure you run the tests from within the Tests folder")
+    sys.exit()
+# alternatively you could use any predefined dictionary like this:
+# from CaiIndices import SharpIndex # you can save your dictionary in this file.
+# X.SetCaiIndex(SharpIndex)
+
+infilename = 'sample_genes.fas'
+outfilename = 'nRCA_test_output.csv'
```

Comment:
This would ideally be a temp file, e.g. via ``import tempfile``, rather than generating output in the ``Tests/`` folder.